### ISSUE TITLE
"Vertex shader compilation failed" when running on Raspberry Pi 3

### ISSUE THREAD

**Issue by vaguerant** [sm64pc/sm64pc#24]
**Describe the bug**
The game seemed to compile without issue on my Raspberry Pi 3 (original model, not the improved + version), but when running the binary, I get the following output:

```
./build/us_pc/sm64.us.f3dex2e.arm 
Loading configuration from 'sm64config.txt'
Config file 'sm64config.txt' not found. Creating it.
Saving configuration to 'sm64config.txt'
Vertex shader compilation failed

Aborted
```
After the first run, I get the following:
```
./build/us_pc/sm64.us.f3dex2e.arm 
Loading configuration from 'sm64config.txt'
option: 'fullscreen', value: 'false'
option: 'key_a', value: '38'
option: 'key_b', value: '51'
option: 'key_start', value: '57'
option: 'key_l', value: '52'
option: 'key_r', value: '54'
option: 'key_z', value: '37'
option: 'key_cup', value: '328'
option: 'key_cdown', value: '336'
option: 'key_cleft', value: '331'
option: 'key_cright', value: '333'
option: 'key_stickup', value: '17'
option: 'key_stickdown', value: '31'
option: 'key_stickleft', value: '30'
option: 'key_stickright', value: '32'
Vertex shader compilation failed

Aborted
```

**To Reproduce**
1. Compile with `TARGET_RPI` (I simply built the US version)
2. Attempt to launch the resulting binary (from terminal)
3. See output.

**Expected behavior**
I don't know about "expected" since I'm not sure if Pi building is considered ready, I really just wanted to see if it would work, but the *ideal* behavior would probably be that the game would work. :)

**Desktop (please complete the following information):**
 - OS: Raspbian
 - Version: 9 (Stretch)

**Additional context**
I tried compiling with and without `TARGET_BITS=64`.

**Comment by IvanDSM** [comment]
Which branch did you try to compile?

**Comment by Hyenadae** [comment]
Shit. I had this issue too ages ago.

**Comment by vaguerant** [comment]
@IvanDSM Just the master branch.

@Hyenadae

> Make sure you're using VC4 DRM (raspi-config -> adv -> openGL KMS enabled, either one).
> Try to run the program then?

I'm almost certain this is my problem, but I can't test properly as I get no video signal at all from the Pi when running with either the full or fake KMS options. I was able to SSH in and launch the previously compiled executable and it didn't throw the error in the conso!e or abort any more, I assume it was running without issue as it didn't exit until I ended the process myself.

I haven't updated my Raspbian distribution in a couple of years, so I imagine I'd probably have more luck with an update, but I have a lot of things running on the Pi that would be inconvenient to get going on a newer image, so I think I'll just leave this one alone. Thanks very much for the assistance and I apologize for any time wasted by this issue.
